services/ClamAvScanningService.cfc

```diff
@@ -25,17 +41,51 @@ component {
 			report.stdErr        = _processStream( process.getErrorStream() );
 			report.stdOut        = _processStream( process.getInputStream() );
 			report.virusDetected = ( report.exitValue == VIRUS_DETECTED );
+		} catch( any e ) {
+			$raiseError( e );
+			report.virusDetected = false;
+			report.stdErr        = "Message: [#e.message#]. Detail: [#e.detail#].";
+		}
 
+		return report.virusDetected;
+	}
+
+	private boolean function _scanExternal( required string filePath, required struct report ) {
+		try {
+			var ClamAvClient     = _getClamAvClient();
+			var inputStream      = CreateObject( "java", "java.io.FileInputStream" ).init( arguments.filePath );
+			var scanResult       = ClamAvClient.scan( inputStream );
+
+			report.stdOut        = _processExternalResponse( scanResult );
+			report.virusDetected = !ClamAvClient.isCleanReply( scanResult );
 		} catch( any e ) {
 			$raiseError( e );
 			report.virusDetected = false;
-			report.stdErr = "Message: [#e.message#]. Detail: [#e.detail#].";
+			report.stdErr        = "Message: [#e.message#]. Detail: [#e.detail#].";
 		}
 
 		return report.virusDetected;
 	}
 
+	public boolean function useExternalClamAv() {
+		return Len( _getExternalHostname() ) > 0;
+	}
+
 // PRIVATE HELPERS
+	private any function _getClamAvClient() {
+		return CreateObject( "java", "fi.solita.clamav.ClamAVClient", _getLib() ).init( _getExternalHostname(), _getExternalPort() );
+	}
+
+	private array function _getLib() {
+		_lib = _lib ?: DirectoryList( ExpandPath( GetDirectoryFromPath( GetCurrentTemplatePath() ) & "lib" ), false, "path" );
```

Comment:
We found rather unexpected results when using `GetCurrentTemplatePath()` like this (i.e. when a project chooses to extend this service, this will break). Just this should be fine:

```cfc
ExpandPath( "/app/extensions/preside-ext-clamav/services/lib"  )
```